From 4ba461b7c96ede2ff60d7f62393500684b246bf8 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Mon, 9 Mar 2015 13:56:53 +0800
Subject: [PATCH] MLK-10387-1 ARM: imx7d: set dr_mode for "host" if controller
 is host-only

The dts is used to describe hardware capabilities, so it is
better to add dr_mode = "host" if the controller is host-only,
it can also eliminate "doesn't support gadget" message for host-only
controller initialization. Below thread also discuss it:

http://lkml.iu.edu/hypermail/linux/kernel/1502.3/02871.html

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 arch/arm/boot/dts/imx7d.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index 77f29b5..908f5993e 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -1017,6 +1017,7 @@
 				fsl,usbphy = <&usbphy_nop3>;
 				fsl,usbmisc = <&usbmisc3 0>;
 				phy_type = "hsic";
+				dr_mode = "host";
 				status = "disabled";
 			};
 
-- 
1.8.0

